core/include/jiminy/core/robot/Robot.h

```diff
@@ -65,6 +66,38 @@ namespace jiminy
                             Eigen::Ref<vectorN_t const> const & v,
                             Eigen::Ref<vectorN_t const> const & a,
                             vectorN_t                   const & u);
+
+        /// \brief Add a kinematic constraint to the robot.
+        ///
+        /// \param[in] constraintName Unique name identifying the kinematic constraint.
+        /// \param[in] constraint Constraint to add.
+        hresult_t addConstraint(std::string const & constraintName,
+                                std::shared_ptr<AbstractConstraint> constraint);
+
+        /// \brief Remove a kinematic constraint form the system.
+        ///
+        /// \param[in] constraintName Unique name identifying the kinematic constraint.
+        hresult_t removeConstraint(std::string const & constraintName);
```

Comment:
.

Reply:
I know, but is this really something we need ? The constraint is kind of passive to me: it's there, but there isn't much things one would want to do with it beside creating it.

Reply:
It could be useful for debug a step-by-step simulation, when the contraint is not accessible because it has been created by another method (as we are doing for atalante). Getting access to the contraint is necessary to extract the Jacobian,.

Reply:
True, adding function.